nilearn/decoding/decoder.py

```diff
@@ -106,41 +107,131 @@ def _check_param_grid(estimator, X, y, param_grid=None):
 
     """
     if param_grid is None:
+        param_grid = _default_param_grid(estimator, X, y)
+
+    else:
+        if isinstance(estimator, (RidgeCV, RidgeClassifierCV)):
+            param_grid = _wrap_param_grid(param_grid, "alphas")
+
+    return param_grid
+
+
+def _default_param_grid(estimator, X, y):
+    """Generate sensible default for param_grid.
+
+    Parameters
+    ----------
+    estimator: str, optional
+        The estimator to choose among:
+        %(classifier_options)s
+        %(regressor_options)s
+        Default 'svc'.
+
+    X: list of Niimg-like objects
+        See :ref:`extracting_data`.
+        Data on which model is to be fitted. If this is a list,
+        the affine is considered the same for all.
+
+    y: array or list of shape (n_samples)
+        The dependent variable (age, sex, IQ, yes/no, etc.).
+        Target variable to predict. Must have exactly as many elements as
+        3D images in niimg.
+
+    Returns
+    -------
+    param_grid: dict of str to sequence, or sequence of such. Sensible default
+    dict has size 1 for linear models.
+    """
+    param_grid = {}
+
+    # define loss function
+    if isinstance(estimator, LogisticRegression):
+        loss = "log"
+    elif isinstance(estimator, (LinearSVC, RidgeCV, RidgeClassifierCV, SVR)):
+        loss = "squared_hinge"
```

Comment:
Why for Ridge ?

Reply:
@bthirion this was from the existing code, I just moved it to another function as per @jeromedockes's suggestion. But you're right, it doesn't seem to make much sense: `loss` is only used in the `if` block on Lines 163-164, but `RidgeCV`, `RidgeClassifierCV` and `SVR` all don't have the `penalty` attribute, so it looks like this is useless. I can change it here, or maybe we should have another PR for it? Since this PR is supposed to address wrapping of the `param_grid` for `RidgeCV`/`RidgeClassifierCV`

Reply:
as you prefer! as you're improving the handling of the parameter grid and you're touching that code anyway, I think you can do this cleanup in the same PR